app/lib/actions/katello/event_queue/poller_thread.rb

```diff
@@ -0,0 +1,44 @@
+module Actions
+  module Katello
+    module EventQueue
+      class PollerThread
+        attr_reader :instance
+
+        def self.initialize(logger)
+          @instance ||= self.new(logger)
+        end
+
+        def self.close
+          @instance.close if @instance
+          @instance = nil
+        end
+
+        def initialize(logger)
+          @logger = logger
+        end
+
+        def close
+          @thread.kill if @thread
+        end
+
+        def poll_for_events(suspended_action)
+          @thread.kill if @thread
+          @thread = Thread.new do
+            loop do
+              begin
+                until (event = ::Katello::EventQueue.peek_event).nil?
+                  suspended_action.notify_queue_item(event.event_type, event.object_id, event.created_at) if event
+                end
+
+                sleep 2
```

Comment:
hah,  5 seemed too long, 1 seemed too short.  I'm open to something different ;)